worker/docker_client.py

```diff
@@ -174,23 +174,24 @@ def start_container(self, bundle_path, uuid, command, docker_image,
             ]
         docker_commands = libcuda_commands + [
             'ldconfig',
-            'BASHRC=$(pwd)/.bashrc',
-            # Run as the user that owns the bundle directory. That way
-            # any new files are created as that user and not root.
             'U_ID=$(stat -c %%u %s)' % docker_bundle_path,
             'G_ID=$(stat -c %%g %s)' % docker_bundle_path,
-            'sudo -u \\#$U_ID -g \\#$G_ID -n bash -c ' +
-            # We pass several commands for bash to execute as the user as a
-            # single argument (i.e. all commands appear in quotes with no spaces
+            'BASHRC=$(pwd)/.bashrc',
+            # We pass several commands for bash to execute as a single
+            # argument (i.e. all commands appear in quotes with no spaces
             # outside the quotes). The first commands appear in double quotes
             # since we want environment variables to be expanded. The last
-            # appears in single quotes since we do not. The expansion there, if
-            # any, should happen when bash executes it. Note, since the user's
-            # command can have single quotes we need to escape them.
-            '"[ -e $BASHRC ] && . $BASHRC; "' +
-            '"cd %s; "' % docker_bundle_path +
-            '"export HOME=%s; "' % docker_bundle_path +
-            '\'(%s) >stdout 2>stderr\'' % command.replace('\'', '\'"\'"\''),
+            # appears in single quotes since we do not. The expansion there,
+            # if any, should happen when bash executes it. Note, since the
+            # user's command can have single quotes we need to escape them.
+            'bash -c '
+            + '"[ -e $BASHRC ] && . $BASHRC; "'
+            + '"cd %s; "' % docker_bundle_path
+            + '"export HOME=%s; "' % docker_bundle_path
+            + '\'(%s) >stdout 2>stderr\'' % command.replace('\'', '\'"\'"\''),
+            # Ensure that any created files are owned by the user/group that
+            # owns the bundle directory, not root.
+            'chown -R $U_ID:$G_ID %s' % docker_bundle_path,
```

Comment:
I had an earlier solution that created the Docker container to run as the proper user/group, but then running `ldconfig` didn't work because it required root permissions. And `su` complained that it has to be run "in a terminal." Still trying to untangle this..